camelcase-keys/index.d.ts

```diff
@@ -0,0 +1,7 @@
+// Type definitions for camelcase-keys 4.0
+// Project: https://github.com/sindresorhus/camelcase-keys#readme
+// Definitions by: Mohamed Hegazy <https://github.com/mhegazy>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = camelcase_keys;
+declare function camelcase_keys(input: any, options?: { exclude?: string[] | RegExp[], deep?: boolean }): any;
```

Comment:
Looks like we'll infer a generic type correctly even for non-homogenous objects:
```ts
declare function f<T>(obj: { [key: string]: T }): { [key: string]: T };
const o = f({ x: 0, y: "" }); // o: { [key: string]: string | number; }
```